### ISSUE TITLE
[Accessibility Audit] Insights: Discover insights page

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#33492]
### Steps to replicate journey

Taken from the [user journey list document](https://docs.google.com/document/d/1m8oYTFgim8epA2IfPFCTN2QFSDxODV5WJNe2JhRdMZA/edit?usp=sharing). Use this for further context.

- Open the main sourcegraph page (sourcegraph.com)
- Click on the “Insights” navigation tab
- See the dashboard page with “code insights” tab or “getting started” tab
- Change active dashboard with dashboard select

### How to audit

Follow the instructions here: [Auditing a user journey](https://docs.sourcegraph.com/dev/background-information/web/accessibility/how-to-audit#auditing-a-user-journey)

**Note:** We are planning on arranging thorough third-party audit in the future, so our focus here is just to find the _majority_ of accessibility issues. Don't worry if you feel you haven't found 100% of potential issues, it is better to focus on the core essentials to complete the journey rather than spending lots of time going through every possible problem.

### How to raise discovered problems

Follow the instructions here: [Raising an accessibility bug](https://docs.sourcegraph.com/dev/background-information/web/accessibility/how-to-audit#raising-a-bug)

### Any other questions?

See the [audit FAQ](https://docs.google.com/document/d/1nCi7TvYRpj9eG0_QtgqK1c4L28EiJKVuxyuvg1lAGgo/edit#) - add your question if it isn't there already!

**Comment by sourcegraph-bot-2** [comment]
Heads up @joelkw @felixfbecker @vovakulikov @unclejustin - the "team/code-insights" label was applied to this issue.

**Comment by felixfbecker** [comment]
@vovakulikov do we have to do this before 4.0? If yes, could somebody else than you do this audit?

**Comment by taylorsperry** [comment]
This work can wait until after 4.0, as we're pausing the accessibility effort until after the launch: https://sourcegraph.slack.com/archives/C038K3PR2M7/p1658530450520339

**Comment by felixfbecker** [comment]
@sourcegraph/code-insights-frontend  I'm noticing a bunch of high-level usability showstoppers when using a screen reader on the dashboard page that don't have any issue currently:

- Every card is simply labelled "Insights card, region", meaning a screen reader user does not actually have a way to understand semantically what insights are on the dashboard by title. The heading/title itself cannot be read either, because it is a link and has the `aria-label` "Go to insights card". This also leads to hundreds of "Insights card" reported in landmarks and "Go to insights card" links without a descriptive title. The card itself should announce the title (label) of the insight.
- The grid elements are not marked up as a grid/grid items, meaning there is no way to navigate them spatially or orient yourself in them. They are all just regions. Related to my comment on https://github.com/sourcegraph/sourcegraph/issues/41680.
